docs/adr/0036-specifying-partial-segment-usage.md

```diff
@@ -0,0 +1,62 @@
+* TAMS is intended to support a variety of media container formats, so the solution should be generally applicable
+* Theoretically partial audio segments can be selected on individual samples, not just coded frames
+
+## Considered Options
+
+* Option 1: Retain both sample-based and time-based selection mechanism
+* Option 2: Remove sample-based mechanism
+
+## Decision Outcome
+
+Chosen option: TBD, because
+{Justification, e.g., only option which resolves requirements, or comes out best (see below)}.
+
+### Implementation
+
+{Once the proposal has been implemented, add a link to the relevant PRs here}
+
+## Pros and Cons of the Options
+
+### Option 1: Retain both sample-based and time-based selection mechanism
+
+Retain the existing combination of both `sample_offset`/`sample_count` along with the ability to calculate grains to use vs discard based on `timerange` and `ts_offset`.
+Stipulate throughout the documentation that if the defaults (`sample_offset=0` and `sample_count` is the entire segment) are not valid because part of the segment has been used, then they must be set correctly.
+Call out in documentation that either method can be used by a reader.
+
+* Good, because it may be easier to select the correct grains by counting and discarding (when temporal re-ordering is not applied).
+  For example it may be possible to avoid parsing/demuxing the container to extract timestamps, if Flow timing is carried out-of-band
+* Good, because it avoids a change to the existing API
+* Neutral, because it works for containers without any internal timing where the Flow doesn't have a rate, however there are no current known use cases (or containers!) with this property
+* Bad, because a writer has to do additional work to identify the correct `sample_offset`/`sample_count` settings when re-using segments, which may require it to download and parse the media
+* Bad, because this is even more difficult if grains have variable duration/gaps and are out of order: a writer must first switch them to presentation order, then filter on the desired TimeRange and count
+
+### Option 2: Remove the sample-based selection mechanism
+
+Remove `sample_offset` and `sample_count` from the API (possibly by first deprecating).
+Make partial-segment re-use via `timerange`/`ts_offset` clearer in the documentation.
+Readers will then need to calculate the timestamp for each grain on the Flow timeline by applying `ts_offset` to the media object's internal timeline, then check whether that timestamp is contained by the segment's `timerange`.
+
+* Good, because it avoids un-necessary work for writers of partial-segment Flows
+* Good, because it removes duplicated information and potential for an error condition
+* Neutral, because it forces the use of either containers with some kind of internal timing or Flows with a rate and no gaps
```

Comment:
I think it only requires Objects with no gaps, when there's no internal timing? Gaps between segments should be fine?